### PR TITLE
Update issue key generation

### PR DESCRIPTION
**Pull request by R-Mielamud**

### CODE REVIEW

File: client/src/components/IssueFileInput/index.tsx [hunk 1]
```diff
@@ -13,80 +11,90 @@ import {
...
+	};
+
+	const curryDeleteLink = (index: number) => () => {
+		const newLinks = [...alreadyAttached];
 		newLinks.splice(index, 1);
-		onChange(newLinks);
+		onDeleteAlreadyAttached(newLinks);
+	};
+
+	const Body: React.FC = () => {
```
Comment (sahanr) [inline client/src/components/IssueFileInput/index.tsx:53]: move it to separate component

File: server/src/repositories/issue.repository.ts [hunk 2]
```diff
@@ -49,6 +52,17 @@ export type Filter = {
 	comment?: string;
 };
 
+export interface Create {
```
Comment (sahanr) [inline server/src/repositories/issue.repository.ts:55]: rename to CreateIssueArgs

File: server/src/repositories/issue.repository.ts [hunk 3]
```diff
@@ -88,9 +102,24 @@ export class IssueRepository extends Repository<Issue> {
...
 
-	async createOne(data: Issue) {
-		const entity = this.create(data);
-		const result = await this.save(entity);
+	async createOne(data: Create) {
+		const { project } = data;
+		const projectReposritory = getCustomRepository(ProjectsRepository);
+		const { key, issues = [] } = (await projectReposritory.getWithIssuesById(project)) as Projects;
+		const e = extractIndexFromIssueKey;
+		const sorted = issues.sort((i0, i1) => (e(i0.issueKey as string) > e(i1.issueKey as string) ? 1 : -1));
```
Comment (sahanr) [inline server/src/repositories/issue.repository.ts:110]: use reduce func

**Review by sahanr — APPROVED** [review]

File: client/src/components/IssueFileInput/index.tsx (hunk 1 above)
Comment (R-Mielamud) [reply client/src/components/IssueFileInput/index.tsx:53]: I can't it uses many props and functions from IssueFileInput
I will move it to render

**Pull request merged**